### PR TITLE
Adding get Quotes to simulator

### PR DESCRIPTION
**Pull request by dnamkaa**
getQUotes handler has been added to enable simulator to get quotes by Id.

### CODE REVIEW

File: src/payee/handler.js [hunk 1]
```diff
@@ -30,9 +30,9 @@ const Logger = require('@mojaloop/central-services-shared').Logger
 const Metrics = require('../lib/metrics')
 const base64url = require('base64url')
 
-const partiesEndpoint = process.env.PARTIES_ENDPOINT || 'http://localhost:1080'
-const quotesEndpoint = process.env.QUOTES_ENDPOINT || 'http://localhost:1080'
-const transfersEndpoint = process.env.TRANSFERS_ENDPOINT || 'http://localhost:1080'
+const partiesEndpoint = process.env.PARTIES_ENDPOINT || 'http://localhost:8444'
+const quotesEndpoint = process.env.QUOTES_ENDPOINT || 'http://localhost:3002'
```
Comment (rmothilal) [inline src/payee/handler.js]: Please switch this back to the original ports as this is configured

File: src/payee/handler.js [hunk 2]
```diff
@@ -167,12 +167,102 @@ exports.getPartiesByTypeAndId = function (req, h) {
...
+  })()
+  return h.response().code(200)
+}
 
 exports.postQuotes = function (req, h) {
   (async function () {
     const histTimerEnd = Metrics.getHistogram(
       'sim_request',
-      'Histogram for Simulator http operations',
+      'Histogram for Simula tor http operations',
```
Comment (rmothilal) [inline src/payee/handler.js]: Please fix spelling

File: src/payee/handler.js [hunk 3]
```diff
@@ -167,12 +167,102 @@ exports.getPartiesByTypeAndId = function (req, h) {
...
+        amount: '1',
+        currency: quotesRequest.amount.currency
+      },
+      payeeFspCommission: {
+        amount: '1',
+        currency: quotesRequest.amount.currency
+      },
+      expiration: quotesRequest.expiration,
+
+      // ilpPacket: 'AQAAAAAAAABkEHByaXZhdGUucGF5ZWVmc3CCAlV7InRyYW5zYWN0aW9uSWQiOiJhYWUwYzIxMi0wOTJiLTQ5MmItYWQ2ZS1kZmJiYmJjNWRkYzIiLCJxdW90ZUlkIjoiYWFlMGMyMTItMDkyYi00OTJiLWFkNmUtZGZiYmJiYzVkZGMyIiwicGF5ZWUiOnsicGFydHlJZEluZm8iOnsicGFydHlJZFR5cGUiOiJNU0lTRE4iLCJwYXJ0eUlkZW50aWZpZXIiOiIyMjUwNDAwNDc2MiIsImZzcElkIjoicGF5ZWVmc3AifSwicGVyc29uYWxJbmZvIjp7ImNvbXBsZXhOYW1lIjp7fX19LCJwYXllciI6eyJwYXJ0eUlkSW5mbyI6eyJwYXJ0eUlkVHlwZSI6Ik1TSVNETiIsInBhcnR5SWRlbnRpZmllciI6IjI3NzEzODAzOTA1IiwiZnNwSWQiOiJwYXllcmZzcCJ9LCJwZXJzb25hbEluZm8iOnsiY29tcGxleE5hbWUiOnsiZmlyc3ROYW1lIjoiTWF0cyIsImxhc3ROYW1lIjoiSGFnbWFuIn19fSwiYW1vdW50Ijp7ImN1cnJlbmN5IjoiVVNEIiwiYW1vdW50IjoiMTAwIn0sInRyYW5zYWN0aW9uVHlwZSI6eyJzY2VuYXJpbyI6IlRSQU5TRkVSIiwic3ViU2NlbmFyaW8iOiJUUkFOU0ZFUiIsImluaXRpYXRvciI6IlBBWUVSIiwiaW5pdGlhdG9yVHlwZSI6IkNPTlNVTUVSIiwicmVmdW5kSW5mbyI6e319LCJub3RlIjoiaGVqIn0=',
```
Comment (rmothilal) [inline src/payee/handler.js]: Please remove commented out code

File: src/payee/handler.js [hunk 4]
```diff
@@ -167,12 +167,102 @@ exports.getPartiesByTypeAndId = function (req, h) {
...
+          'FSPIOP-URI': `/quotes/${quotesRequest.quoteId}`
+        },
+        rejectUnauthorized: false,
+        body: JSON.stringify(quotesResponse)
+      }
+      Logger.info((new Date().toISOString()), `Executing PUT`, url)
+      const res = await fetch(url, opts)
+      Logger.info((new Date().toISOString()), 'response: ', res.status)
+      if (!res.ok) {
+        // TODO: how does one identify the failed response?
```
Comment (rmothilal) [inline src/payee/handler.js]: Please remove todo or compete it

File: src/payee/handler.js [hunk 5]
```diff
@@ -167,12 +167,102 @@ exports.getPartiesByTypeAndId = function (req, h) {
...
+      // Logger.perf(`[cid=${req.payload.transferId}, fsp=${req.headers['fspiop-source']}, source=${req.headers['fspiop-source']}, dest=${req.headers['fspiop-destination']}] ~ Simulator::api::payee::postQuotes - END`)
+      histTimerEnd({ success: true, fsp: 'payee', operation: 'getQuotes', source: req.headers['fspiop-source'], destination: req.headers['fspiop-destination'] })
+    } catch (err) {
+      Logger.error(err)
+      // Logger.perf(`[cid=${req.payload.transferId}, fsp=${req.headers['fspiop-source']}, source=${req.headers['fspiop-source']}, dest=${req.headers['fspiop-destination']}] ~ Simulator::api::payee::postQuotes - ERROR`)
+      histTimerEnd({ success: false, fsp: 'payee', operation: 'getQuotes', source: req.headers['fspiop-source'], destination: req.headers['fspiop-destination'] })
+      // TODO: what if this fails? We need to log. What happens by default?
+      // const url = await rq.createErrorUrl(db, req.path, requesterName);
+      // TODO: review this error message
+      // TODO: we should be able to throw an AppError somewhere, test whether the error
```
Comment (rmothilal) [inline src/payee/handler.js]: Please remove all comments

File: src/payee/handler.js [hunk 6]
```diff
@@ -204,7 +294,8 @@ exports.postQuotes = function (req, h) {
     //     data: Buffer.from('hello world'),
     //     expiresAt: new Date(new Date().getTime() + 10000)
```
Comment (rmothilal) [inline src/payee/handler.js]: Remove comments please

**Comment by mdebarros** [comment]
Hi @dnamkaa, are you able to address @rmothilal's comments?

We would very much like to include this in a release \:D/

**Comment by elnyry-sam-k** [comment]
@dnamkaa , please recreate the PR with latest base while addressing any relevant comments above, this is now outdated..

**Pull request closed without merging**